qiskit_nature/second_q/operators/spin_op.py

```diff
@@ -18,532 +18,602 @@
+                "Y_0 Y_1": -1,
+                "Z_0 Z_1": -1,
+                "Z_0": -0.3,
+                "Z_1": -0.3,
+            },
+            spin=1
+        )
 
-    **Initialization**
+    This means :math:`- S^x_0 S^x_1 - S^y_0 S^y_1 - S^z_0 S^z_1 - 0.3 S^z_0 - 0.3 S^z_1`.
 
-    The :class:`SpinOp` can be initialized by the list of tuples.
-    For example,
+    An example using labels with powers would be:
 
     .. jupyter-execute::
 
         from qiskit_nature.second_q.operators import SpinOp
 
-        x = SpinOp("X", spin=3/2)
-        y = SpinOp("Y", spin=3/2)
-        z = SpinOp("Z", spin=3/2)
+        op = SpinOp({"X_0^2 Y_1^3 Z_0": 1})
 
-    are :math:`S^x, S^y, S^z` for spin 3/2 system.
-    Two qutrit Heisenberg model with transverse magnetic field is
+
+    By default, this way of initializing will create a full copy of the dictionary of coefficients.
+    If you have very restricted memory resources available, or would like to avoid the additional
+    copy, the dictionary will be stored by reference if you disable ``copy`` like so:
 
     .. jupyter-execute::
 
-        SpinOp(
-            [
-                ("XX", -1),
-                ("YY", -1),
-                ("ZZ", -1),
-                ("ZI", -0.3),
-                ("IZ", -0.3),
-            ],
-            spin=1
+        some_big_data = {
+            "X_0 Y_0": 1.0,
+            "X_1 Y_1": -1.0,
+            # ...
+        }
+
+        op = SpinOp(
+            some_big_data,
+            num_spins=2,
+            copy=False,
         )
 
-    This means :math:`- S^x_0 S^x_1 - S^y_0 S^y_1 - S^z_0 S^z_1 - 0.3 S^z_0 - 0.3 S^z_1`.
 
-    :class:`SpinOp` can be initialized with internal data structure (`numpy.ndarray`) directly.
-    In this case, `data` is a tuple of two elements: `spin_array` and `coeffs`.
-    `spin_array` is 3-dimensional `ndarray`. 1st axis has three elements 0, 1, and 2 corresponding
-    to x, y, and z.  2nd axis represents the index of terms.
-    3rd axis represents the index of register.
-    `coeffs` is one-dimensional `ndarray` with the length of the number of terms.
+    .. note::
+
+        It is the users' responsibility, that in the above scenario, :code:`some_big_data` is not
+        changed after initialization of the `SpinOp`, since the operator contents are not
+        guaranteed to remain unaffected by such changes.
 
     **Algebra**
 
     :class:`SpinOp` supports the following basic arithmetic operations: addition, subtraction,
-    scalar multiplication, and adjoint.
+    scalar multiplication, adjoint, composition and tensoring.
+
+    As of now, operations that involve two different instances of `SpinOp` (i.e. addition,
+    subtraction, composition, and tensoring) are only supported for identical spins
+    (`op_1.num_spins == op_2.num_spins`). In the future, multiple spin lengths within one
+    `SpinOp` will be supported, which might enable certain operations to also support
+    different `num_spins`.
```

Comment:
We should remove the second sentences. We should not promise any features (even if they are _very_ likely).
Let's simply state that only identical spin operations are supported (and then in the future we rephrase this)